### PR TITLE
#inv:Implement Category endpoint

### PR DESCRIPTION
**Pull request by cnangeh**

### CODE REVIEW

File: src/main/java/com/tragent/inventory/controllers/CategoryController.java [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+		
+	}
+	
+	/**
+	 * Delete category's information
+	 * 
+	 * @param id, category's id
+	 * @return 204, .
+	 */
+	@RequestMapping(value="/{id}",
```
Comment (awasum) [inline src/main/java/com/tragent/inventory/controllers/CategoryController.java:115]: Are we still deleting?

File: src/main/java/com/tragent/inventory/model/Category.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.Id;
+import javax.persistence.Table;
+
+@Entity
+@Table(name = "category")
+public class Category {
```
Comment (awasum) [inline src/main/java/com/tragent/inventory/model/Category.java:11]: is category not linked to products??

File: src/main/java/com/tragent/inventory/controllers/CategoryController.java [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+	 */
+	@RequestMapping(method=RequestMethod.GET, 
+			produces = MediaType.APPLICATION_JSON_VALUE)
+	public ResponseEntity<Collection<Category>> getUsers(@RequestParam(value = "name", required = false) String name){
+		
+		Collection<Category> categories = new ArrayList<Category>();
+		if (name != null) {
+			Category category = categoryService.findByName(name);
+			categories.add(category);
+		} else {
```
Comment (awasum) [inline src/main/java/com/tragent/inventory/controllers/CategoryController.java:40]: Ok, looks good.

File: src/main/java/com/tragent/inventory/model/Category.java (hunk 2 above)
Comment (DanCarl857) [reply src/main/java/com/tragent/inventory/model/Category.java:11]: She can link this to the products(that is do it as a subresource) or vice versa once they have implemented the products endpoint

File: src/main/java/com/tragent/inventory/service/implementation/CategoryServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+			return null;
+		}
+		
+		categoryRepository.save(category);
+		return category;
+		
+	}
+
+	@Override
+	public void delete(Long id) {
```
Comment (awasum) [inline src/main/java/com/tragent/inventory/service/implementation/CategoryServiceImpl.java:67]: Good, this just disables.

**Pull request merged**